### PR TITLE
Update node version to v17.9 and switch from node-sass to sass

### PR DESCRIPTION
**Pull request by devenini**
This pull request updates the node version from `v14.17` to `v17.9`, and replaces the deprecated node-sass module with sass. These updates will simplify maintenance and future-proof the boilerplate by ensuring the use of non-deprecated packages.

Key changes : 

- The use of SASS syntax for math functions.
- The removal of the app-env() function from the node-sass module, as it was not used. Instead, inline styles are used for development scoped modules.

### CODE REVIEW

**Comment by devenini** [comment]
Node 18 is not available in Cloudflare Pages for now but will be in the near future.
[Source](https://community.cloudflare.com/t/pages-build-failed-when-using-node-18-12-1-due-to-missing-glibc/433516)

An easy quick fix is to downgrade the node version to `v.17.x` for projects with an environment that requires a specific version.

**Comment by mcaskill** [comment]
The [sass](https://www.npmjs.com/package/sass) package is a distribution of [Dart Sass](https://github.com/sass/dart-sass) compiled to pure JavaScript.

For those who have Dart Sass with Dart VM, is there a way to ignore the NPM package and use the native binary?

I have not tested how fast Dart is compared to the JS compiled version.

**Comment by devenini** [comment]
To my knowledge, it is not possible to ignore the NPM package and use the native binary if Dart Sass is already installed. However, using the JavaScript library provides a convenient way to manage dependencies with NPM and aligns with our other JavaScript build tasks. Additionally, prioritizing the use of NPM packages is important, especially as we are using more cloud-based deployment solutions like Cloudflare Pages.

**Review by mcaskill — APPROVED** [review]

File: build/utils/config.js [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
-    usrconfig = await import('../../loconfig.local.json');
+    usrconfig = await import('../../loconfig.local.json', {
+        assert: { type: 'json' }
+    });
     usrconfig = usrconfig.default;
 
     merge(loconfig, usrconfig);
 } catch (err) {
-    // do nothing
+    console.error(err);
```
Comment (mcaskill) [inline build/utils/config.js:17]: Does this log an error if you don't have a local config file?

Comment (devenini) [reply build/utils/config.js:17]: It does, good catch. I will revert the change on this line

**Pull request merged**